src/services/game-logic/GameEngine.ts

```diff
@@ -10,7 +12,16 @@ class GameEngine {
 
     getLegalMoves = (square: Square): Square[] => {
         const piece = this.board.getPiece(square);
-        return piece?.getPossibleMoves() ?? [];
+        let pieceMoves = piece?.getPossibleMoves() ?? [];
+        return pieceMoves.filter((square) => this.isOccupiedBySameColorPiece(square, piece));
+    };
+
+    private isOccupiedBySameColorPiece = (square: Square, piece: Piece | null): Boolean => {
+        if (piece?.name === PieceNames.PAWN) {
```

Comment:
dlaczego tylko na pionki?

Reply:
bo pionki biją inaczej niż reszta, a więc jeśli bierka może wykonać legalny ruch, czyli zgodnie ze swoimi zasadami, a na dostępnym polu stoi bierka przeciwnika, to ją bije, no chyba że jest pioniek, to nie bije i nie może ta stanąć.
bicie przez pionka to inne zadanie